### ISSUE TITLE
Allow selection of AD tenants in making azure core API calls. 

### ISSUE THREAD

**Issue by aasimkhan30** [microsoft/azuredatastudio#14644]
Currently all the queries exposed by azure core run on all the tenants that are available in the passed account. Ideally, the callers should be able to select a tenant or a set of tenants to run these queries on.

**Comment by kenvanhyning** [comment]
Definitely should do this but doesn't meet ask mode bar.  Postponing - please bring back for May if the code is ready.

**Comment by watahani** [comment]
Hi @aasimkhan30  and @kenvanhyning 

The inability to select a tenant has caused the following problems;

## Situations

- an user whose home tenant is Contoso is invited Fabrikam tenant.
- the user have valid permission to access SQL Server in **Fabrikam** tenant via AAD Authentication.
- the user is blocked to access Contoso subscription by Conditional Access Policy but not Fabrikam tenant.

## Issue

User cannot access the SQL Server in Fabrikam tenant using Azure Data Studio, because Azure Data Studio using common endpoint for acquiring tokens. If the user access common AuthZ endpoint, he is blocked by Conditional Access Policy in Contoso tenant.

```http
https://login.microsoftonline.com/common/oauth2/authorize?response_type=code&response_mode=query&client_id=a69788c6-1d43-44ed-9ca3-b83e194da255&redirect_uri=https%3a%2f%2fvscode-redirect.azurewebsites.net%2f&state=59026%2chKmfmPv6cwNF%252B623oOC52A%253D%253D&prompt=select_account&code_challenge_method=S256&code_challenge=6gy2tBQXcQlPjb87axh38QYMwy-i1n7OXzV7X26toOc&resource=https%3a%2f%2fmanagement.core.windows.net%2f&sso_nonce=AwABAAAAAAACAOz_BAD0__KatDXloZVd5-TelUD0o_fdkrzrzId09_P101nWj8ffwtTYFZf4wiTf2HDI8rScmyKE8T18zRD_hUyU_ld2mf8gAA&client-request-id=020e3ad6-daa0-4a98-a010-4b1b46e90462&mscrid=020e3ad6-daa0-4a98-a010-4b1b46e90462
```

On the other hand, he can access a SQL Server in Fabrikam tenant using SSMS because SSMS send authorize request with tenant ID:

```http
https://login.microsoftonline.com/{Fabrikam_tenant_id}/oauth2/authorize?resource=https%3A%2F%2Fdatabase.windows.net%2F&client_id=a94f9c62-97fe-4d19-b06d-472bed8d2bcf&response_type=code&haschrome=1&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient&...
```

I would be happy if Azure Data Studio can access to an tenant I selected.

**Comment by cheenamalhotra** [comment]
This has been addressed with Tenant dropdown introduced in v1.45 (upcoming) and all tenants need not be authenticated to when accessing Azure tree resources. Insiders build contains the fix.
